components/CreateForm.js

```diff
@@ -1,57 +1,142 @@
 import { useState, useRef } from "react";
 import styled from "styled-components";
 import { Button, ButtonContainer } from "./Buttons";
+import { TbCheck, TbPlus } from "react-icons/tb";
+import { nanoid, customAlphabet } from "nanoid";
+
+const slugSuffix = customAlphabet(
+  "23456789abcdefghklmnpqrstuvwxyzABCDEFGHKLMNPQRSTUVWXYZ",
+  4
+);
 
 function CreateForm() {
-  const [steps, setSteps] = useState(1);
   const buttonRef = useRef();
+  const [inputSteps, setInputSteps] = useState([
+    { step: 1, stepTitle: "", stepUrl: "", stepDescription: "" },
+  ]);
+  const [inputTutorialTitle, setInputTutorialTitle] = useState("");
 
-  function scrollToButton() {
-    setTimeout(() => {
-      buttonRef.current.scrollIntoView({ behavior: "smooth" });
-    });
+  function handleTitleChange(event) {
+    let titleInput = event.target.value;
+    // Thank you, https://github.com/Roland-Hufnagel
+    titleInput = titleInput.startsWith(" ")
+      ? titleInput.replace(" ", "")
+      : titleInput.includes("  ")
+      ? titleInput.replace("  ", " ")
+      : titleInput;
+    setInputTutorialTitle(titleInput);
+  }
+
+  function handleFormChange(index, event) {
+    const data = [...inputSteps];
+    if (event.target.value.trim() === "") {
+      data[index][event.target.name] = "";
+    } else if (event.target.value[0] === " ") {
+      data[index][event.target.name] = event.target.value.trim();
+    } else {
+      data[index][event.target.name] = event.target.value;
+    }
```

Comment:
This does exactly the same thing as your ternary operators above. Suggestion: create function called sanitizeString which does the sanitation as i described above and use it here as well!